### PR TITLE
Added API key requirement for using the API.

### PR DESCRIPTION
**Pull request by kyleburgess2025**
This PR requires all requests to have an API key. It checks the vector store to see if the owner of the key has the required permissions to use a route.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -124,13 +126,13 @@ async def custom_route_handler(request: Request) -> Response:
 router = APIRouter(route_class=LoggingRoute)
 
 
-@router.get("/")
+@router.get("/", dependencies=[Depends(get_role)])
```
Comment (langdonholmes) [inline src/main.py:129]: Can we leave this route open?

File: src/auth.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from fastapi import Security, HTTPException, status
+from fastapi.security import APIKeyHeader
+from .connections.vectorstore import get_vector_store
+
+api_key_header = APIKeyHeader(name="API-Key")
+
+
+def get_role(api_key_header: str = Security(api_key_header)):
```
Comment (langdonholmes) [inline src/auth.py:8]: I wonder if there is some method to keep the key cached? This will introduce a lot of database calls.

**Review by langdonholmes — CHANGES_REQUESTED** [review]
Kyle, this looks great.

I did some quick research on caching API keys, and a reasonable solution seems to be a "time-to-live" cache. Unfortunately, this will likely require another dependency.

The following looks good to me: [cache tools](https://cachetools.readthedocs.io/en/latest/?ref=vidavolta.io#cachetools.TTLCache)

**Comment by kyleburgess2025** [comment]
@langdonholmes I've made the changes you requested, but I want to make 100000% sure it's working - could you try testing on your end?

**Review by langdonholmes — APPROVED** [review]
Looks great. Let me know if I should deploy this.

**Pull request merged**